arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1936 @@
+              "$ref": "#/definitions/Object"
+            }
+          },
+          "202": {
+            "description": "Accepted",
+            "schema": {
+              "$ref": "#/definitions/IotHubDescription"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/eventHubEndpoints/{eventHubEndpointName}/ConsumerGroups/{name}": {
+      "delete": {
+        "tags": ["DELETE"],
+        "summary": "Delete EventHub Consumer Group for an IotHub",
+        "description": "Delete EventHub Consumer Group for an IotHub",
+        "operationId": "IotHubResource_DeleteEventHubConsumerGroup",
+        "consumes": [],
+        "produces": ["application/json", "text/json"],
+        "parameters": [{
+          "name": "resourceName",
+          "in": "path",
+          "description": "Name of the resource.",
+          "required": true,
+          "type": "string"
+        }, {
+          "name": "eventHubEndpointName",
+          "in": "path",
+          "description": "Name of the event hub endpoint.",
+          "required": true,
+          "type": "string"
+        }, {
+          "name": "name",
+          "in": "path",
+          "description": "The name of the consumer group.",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/subscriptionId"
+        }, {
+          "name": "resourceGroupName",
+          "in": "path",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/api-version"
+        }],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Object"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
+      }
+    },
+    "/providers/Microsoft.Devices/operations": {
+      "get": {
+        "tags": ["GET"],
+        "summary": "Get all supported Resource Provider Operations",
+        "description": "Get all supported Resource Provider Operations",
+        "operationId": "IotHubResource_ListResourceProviderOperations",
+        "consumes": [],
+        "produces": ["application/json", "text/json"],
+        "parameters": [{
+          "$ref": "#/parameters/api-version"
+        }],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ResourceProviderOperationInfoListResult"
+            }
+          },
+          "400": {
```

Comment:
Do you really want to describe the negative respones. Generated code will not throw an Error on 400. It will deserialize based on the schema provided in the 400 response and return the object accordingly. Please take a look at this [doc](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#negative-responses) for more understanding.